GiTinder/Controllers/SettingsController.cs

```diff
@@ -0,0 +1,177 @@
+    {
+        private readonly GiTinderContext _context;
+
+        public SettingsController(GiTinderContext context)
+        {
+            _context = context;
+        }
+
+        //          Mock GET /settings
+        //          Mock PUT /settings
+        //          Tests for all
+
+
+        // GET: Settings
+        //[HttpGet]
+        //[Route("/settings]")]
+
+        [HttpGet("/settings")]
+        public IActionResult GetSettings()
+        {
+            var mockSettings = new Settings() {
+
+                EnableNotification = true;
+            MaxDistanceInKm = 160;
+            PreferredLanguages = "English";
+        }
+        return mockSettings;
+        }
+
+        // GET: Settings
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.Settings.ToListAsync());
+        }
+
+        // GET: Settings/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var settings = await _context.Settings
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (settings == null)
+            {
+                return NotFound();
+            }
+
+            return View(settings);
+        }
+
+        // GET: Settings/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: Settings/Create
+        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
+        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,UserName,EnableNotification,MaxDistanceInKm,PreferredLanguages")] Settings settings)
+        {
+            if (ModelState.IsValid)
+            {
+                _context.Add(settings);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            return View(settings);
+        }
+
+        // GET: Settings/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var settings = await _context.Settings.FindAsync(id);
+            if (settings == null)
+            {
+                return NotFound();
+            }
+            return View(settings);
+        }
+
+        // POST: Settings/Edit/5
+        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
+        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, [Bind("Id,UserName,EnableNotification,MaxDistanceInKm,PreferredLanguages")] Settings settings)
```

Comment:
@FilipTeply in this form unnecessary, this should be the PUT /settings endpoint kinda, but in this format it is not needed